driver-core/src/main/java/com/datastax/driver/core/TypeCodec.java

```diff
@@ -930,6 +933,214 @@ public Date deserialize(ByteBuffer bytes) {
         }
     }
 
+    static class SimpleDateCodec extends TypeCodec<DateWithoutTime> {
+
+        public static final SimpleDateCodec instance = new SimpleDateCodec();
+        private static final Pattern IS_LONG_PATTERN = Pattern.compile("^-?\\d+$");
+
+        private static final String[] patterns = new String[] {
+                                                              "yyyy-MM-dd",
+                                                              "yyyy-MM-ddZ"
+        };
+
+        private SimpleDateCodec() {}
+
+        private static DateWithoutTime parseDate(String str, final String[] parsePatterns) {
+            SimpleDateFormat parser = new SimpleDateFormat();
+            parser.setLenient(false);
+            parser.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+            ParsePosition pos = new ParsePosition(0);
+            for (String parsePattern : parsePatterns) {
+                parser.applyPattern(parsePattern);
+                pos.setIndex(0);
+
+                Date date = parser.parse(str, pos);
+                if (date != null && pos.getIndex() == str.length()) {
+                    return DateWithoutTime.fromMillis(date.getTime());
+                }
+            }
+            throw new IllegalArgumentException("Unable to parse the date: " + str);
+        }
+
+        @Override
+        public DateWithoutTime parse(String value) {
+            if (IS_LONG_PATTERN.matcher(value).matches()) {
+                try {
+                    return DateWithoutTime.fromSimpleDate(Integer.parseInt(value));
```

Comment:
Numeric DATE literals don't fit into an int, because they go from 0 to 2^32 - 1. We need to parse as a long and convert back to an int:
```java
private static final long MAX_LONG_VALUE = (1L << 32) - 1;
private static final long EPOCH_AS_CQL_LONG = (1L << 31);
...
// In CQL, numeric DATE literals are longs between 0 and 2^32 - 1, with the epoch in the middle,
// so parse it as a long and re-center at 0
long cqlLong;
try {
    cqlLong = Long.parseLong(value);
} catch (NumberFormatException e) {
    throw new InvalidTypeException(String.format("Cannot parse date value from \"%s\"", value));
}
if (cqlLong < 0 || cqlLong > MAX_LONG_VALUE)
    throw new InvalidTypeException(String.format("Numeric literals for DATE must be between 0 and %d (got %d)",
        MAX_LONG_VALUE, cqlLong));

int days = (int)(cqlLong - EPOCH_AS_CQL_LONG);
```
Where `cqlLong` is the CQL representation (only positive, epoch is 2147483648), and `days` is our internal representation (normal int ranges, epoch is 0).